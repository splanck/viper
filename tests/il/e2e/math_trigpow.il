il 0.1
extern @rt_print_str(str) -> void
extern @rt_print_f64(f64) -> void
extern @rt_sin(f64) -> f64
extern @rt_cos(f64) -> f64
extern @rt_pow(f64, f64) -> f64

global const str @.nl = "\n"

func @main() -> i64 {
entry:
  %s = call @rt_sin(0.0)
  call @rt_print_f64(%s)
  %nl0 = const_str @.nl
  call @rt_print_str(%nl0)
  %c = call @rt_cos(0.0)
  call @rt_print_f64(%c)
  %nl1 = const_str @.nl
  call @rt_print_str(%nl1)
  %p = call @rt_pow(2.0, 10.0)
  call @rt_print_f64(%p)
  %nl2 = const_str @.nl
  call @rt_print_str(%nl2)
  ret 0
}
