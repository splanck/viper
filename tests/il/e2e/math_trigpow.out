0\n1\n1024\n